src/JuMPContainer.jl

```diff
@@ -208,16 +237,18 @@ function Base.next(x::JuMPArray,k)
     tuple(keys..., x[keys...]), gidx
 end
 
-_next_index{T,N}(x::JuMPArray{T,N}, k) =
+function _next_index{T,N}(x::JuMPArray{T,N}, k)
+    data = printdata(x)
```

Comment:
Rewriting this to eval this definition sees a marginal performance increase for iteration (e.g. 17 secs vs 19 secs)